envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/utils/ReactorUtils.kt

```diff
@@ -47,6 +55,43 @@ fun <T> Flux<T>.onBackpressureLatestMeasured(name: String, meterRegistry: MeterR
         .onBackpressureLatest()
         .measureDiscardedItems(name, meterRegistry)
 
+/**
+ * It is possible that operator combineLatest() or maybe other operators may send cancel() to upstream publisher but
+ * don't send error to a subscriber. The error is stored internally but is not dispatched due to some kind of deadlock.
+ * This method can be used to log such a exception despite that.
+ */
+fun <T> Flux<T>.logSuppressedError(message: String, checkInterval: Duration = defaultCheckInterval): Flux<T> {
+    var subscribtion: Subscription? = null
+    var task: Disposable? = null
+    return this
+        .doOnSubscribe { s ->
+            subscribtion = s
+            task?.dispose()
+            task = scheduler.schedulePeriodically(
+                {
+                    if (logError(s, message)) {
+                        task?.dispose()
```

Comment:
Why should we dispose of this scheduled task? From what I understand we log first error than task is cancelled and  we don't log any further. Is it because ERROR is a terminal operation and there should not be more events?

Reply:
Exactly. If there is an error, nothing else will happen on this flux anynore.